src/components/organisms/UiMenu/UiMenu.vue

```diff
@@ -33,63 +33,38 @@
 
 <script setup lang="ts">
 import { computed } from 'vue';
-import type { PropType } from 'vue';
-import type { HTMLTag } from '../../../types/tag';
-import type { Icon } from '../../../types/icon';
 import UiList from '../UiList/UiList.vue';
+import type { ListAttrsProps } from '../UiList/UiList.vue';
 import UiMenuItem from './_internal/UiMenuItem.vue';
+import type { MenuItemAttrsProps } from './_internal/UiMenuItem.vue';
+import type { DefineAttrsProps } from '../../../types';
 
-export type MenuSuffixVisible = 'default' | 'always' | 'never';
-export interface SuffixAttrs {
-  label?: string;
-  icon?: Icon;
-  iconSuffixAttrs?: Record<string, unknown>;
-  [key: string]: unknown;
-}
-export interface MenuItem {
-  label: string;
+export interface MenuRenderItem extends MenuItemAttrsProps {
   name?: string;
-  icon?: Icon;
-  suffixVisible?: MenuSuffixVisible;
-  suffixAttrs?: SuffixAttrs;
-  buttonMenuItemAttrs?: Record<string, unknown>;
-  [key: string]: unknown;
+  label?: string;
 }
-
-const props = defineProps({
-  /**
-   * Use this to set menu tag.
-   */
-  tag: {
-    type: String as PropType<HTMLTag>,
-    default: 'ul',
-  },
+export interface MenuProps {
   /**
    * Use this props to pass list of menu items.
    */
-  items: {
-    type: Array as PropType<MenuItem[]>,
-    default: () => ([]),
-  },
-});
-const menuItemAttrs = (item: MenuItem) => {
-  /* eslint-disable @typescript-eslint/no-unused-vars */
-  const {
-    name, label, ...rest
-  } = item;
-  /* eslint-enable @typescript-eslint/no-unused-vars */
+  items?: (string | MenuRenderItem)[];
+}
+export type MenuAttrsProps = DefineAttrsProps<MenuProps, ListAttrsProps>;
 
-  return rest;
-};
-const itemsToRender = computed(() => (props.items.map((item, key) => {
+const props = withDefaults(defineProps<MenuProps>(), { items: () => ([]) });
+const menuItemAttrs = ({
+  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+  name, label, ...itemAttrs
```

Comment:
If you don't use `name` and `label` why do you obscure them by decomposition? It's unnecessary and it'll help you with the comment from line 56 :)

Reply:
IMO, we should create a task to add a helper/composable that would filter object keys and return the appropriate attributes for items. WDYT @pspaczek @AlanLes ?